### ISSUE TITLE
register_torch_schedulers incompatible with pytorch-2

### ISSUE THREAD

**Issue by markdjwilliams** [narumiruna/mlconfig#60]
The base class of the learning rate schedulers has [changed](https://github.com/pytorch/pytorch/blob/a7749ae177aa84c60815c869a20772fab7693293/torch/optim/lr_scheduler.py#L169) from `_LRScheduler` to `LRScheduler` in pytorch-2, consequently `mlconfig.torch.register_torch_schedulers()` picks up no schedulers to register.

**Comment by narumiruna** [comment]
Thank you for letting me know this issue.
